config/v1/types_registries.go

```diff
@@ -0,0 +1,47 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Registries holds cluster-wide information about how to handle the registries config.  The canonical name is `cluster`
+type Registries struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	Spec RegistriesSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	Status RegistriesStatus `json:"status"`
+}
+
+type RegistriesSpec struct {
+	// InsecureRegistries are registries which do not have a valid SSL certificate or only support HTTP connections.
+	// +optional
+	InsecureRegistries []string `json:"insecureRegistries,omitempty"`
+	// BlockedRegistries are blacklisted from image pull/push. All other registries are allowed.
+	//
+	// Only one of BlockedRegistries or AllowedRegistries may be set.
+	// +optional
+	BlockedRegistries []string `json:"blockedRegistries,omitempty"`
+	// AllowedRegistries are whitelisted for image pull/push. All other registries are blocked.
+	//
+	// Only one of BlockedRegistries or AllowedRegistries may be set.
+	// +optional
+	AllowedRegistries []string `json:"allowedRegistries,omitempty"`
+}
+
+type RegistriesStatus struct {
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type RegistriesList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Image `json:"items"`
```

Comment:
I looked at the existing build types and if we need this the type for Items should be `[]Registries`. @bparees or @derekwaynecarr can answer if we need `RegistriesList`.

Reply:
Yup, should be []Registries, my bad.

Reply:
I believe we do need `RegistriesList` so that the listers and informers work.

Reply:
Fixed